### PR TITLE
[REVIEW] fix/e2e-concatSuite

### LINKED ISSUE #1583: [BUG] e2e crash running concatSuite
When the e2e concatSuite runs we obtain the following error message:

```shell
===================================================
==>> Run query for test case concatSuite
PLAN:
LogicalProject(c_name=[$4], o_orderkey=[$0], o_orderstatus=[$2])
  PartwiseJoin(condition=[=($1, $3)], joinType=[inner])
    BindableTableScan(table=[[main, orders]], projects=[[0, 1, 2]], aliases=[[o_orderkey, o_custkey, o_orderstatus]])
    BindableTableScan(table=[[main, customer]], filters=[[LIKE(CONCAT('Customer#000000', CAST($0):VARCHAR), 'Customer#0000001')]], projects=[[0, 1]], aliases=[[c_custkey, c_name]])

select c.c_name, o.o_orderkey, o.o_orderstatus
            from orders o
            inner join customer c on o.o_custkey = c.c_custkey
            where CONCAT('Customer#000000', c.c_custkey) like
            'Customer#0000001'


=============== New query: TEST_13 - orc (concatSuite)=================
Steps Complete 0/4|       [20:41:27.258] [error] |||ERROR in kernel::process trying to do do_process. Kernel name is: BindableTableScan Kernel id is: 3|||||                                                    |  0% (00:00 elapsed)
[20:41:27.258] [error] |||ERROR in graph::execute. What: Ral failure at: /blazingsql/sqp/blazingsql-pub/engine/src/execution_kernels/LogicalFilter.cpp:49: Expression does not evaluate to a boolean mask|||||
[20:41:27.265] [error] |||ERROR in graph::execute. What: Ral failure at: /blazingsql/sqp/blazingsql-pub/engine/src/execution_kernels/BatchJoinProcessing.cpp:292: In PartwiseJoin: The right input cache data cannot be null|||||
[20:41:27.265] [error] |||ERROR in graph::execute. What: Ral failure at: /blazingsql/sqp/blazingsql-pub/engine/src/execution_kernels/BatchProcessing.cpp:392: ERROR: Projection::run() first input CacheData was nullptr|||||
Steps Complete 4/4|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████|100% (00:00 elapsed)
Total Batches Processed: 2
[20:41:27.272] [error] 1877255986|||In get_execute_graph_results. What: Ral failure at: /blazingsql/sqp/blazingsql-pub/engine/src/execution_kernels/LogicalFilter.cpp:49: Expression does not evaluate to a boolean mask|||||
==>> Run query for test case concatSuite
PLAN:
LogicalProject(EXPR$0=[CONCAT($2, ' - ', $4)], o_orderstatus=[$1])
  PartwiseJoin(condition=[=($0, $3)], joinType=[inner])
    LogicalProject(o_custkey=[$0], o_orderstatus=[$1], CASE=[CASE(IS NULL($2), 'V1', 'V2')])
      BindableTableScan(table=[[main, orders]], projects=[[1, 2, 0]], aliases=[[o_custkey, o_orderstatus, CASE]])
    BindableTableScan(table=[[main, customer]], filters=[[<($0, 10)]], projects=[[0, 1]], aliases=[[c_custkey, c_name]])

select CONCAT(case when o.o_orderkey is null then 'V1'  else 'V2' end , ' - ', c.c_name),
                  o.o_orderstatus
              from orders o
              inner join customer c on o.o_custkey = c.c_custkey
              where c.c_custkey < 10


=============== New query: TEST_14 - orc (concatSuite)=================
[20:41:27.284] [error] |||ERROR in kernel::process trying to do do_process. Kernel name is: BindableTableScan Kernel id is: 3|||||
[20:41:27.410] [error] |||ERROR in graph::execute. What: Ral failure at: /blazingsql/sqp/blazingsql-pub/engine/src/execution_kernels/LogicalFilter.cpp:49: Expression does not evaluate to a boolean mask|||||
Steps Complete 1/5|█████████████████████████████████████▊```

### PR DESCRIPTION
**Pull request by gcca**
Add e2e for current date and fix some issue to avoid blocking the CI
Fixes #1583

### CODE REVIEW

**Comment by GPUtester** [comment]
Please update the changelog in order to start CI tests.

View the gpuCI docs [here](https://docs.rapids.ai/gpuci#open-pr-or-update-pr).

File: engine/src/parser/expression_utils.cpp [hunk 1]
```diff
@@ -1496,3 +1496,15 @@ std::string preprocess_expression_for_evaluation(std::string expression, blazing
 	expression = apply_interval_conversion(expression, schema);
   return expression;
 }
+
+// This function exists because there was an error performing e2e tests about concat.
+// The function `convert_concat_expression_into_multiple_binary_concat_ops` was removed.
+std::string preprocess_expression_for_filter(std::string expression, blazingdb::manager::Context * context, std::vector<cudf::data_type> schema) {
+	expression = fill_minus_op_with_zero(expression);
+	// expression = convert_concat_expression_into_multiple_binary_concat_ops(expression);
```
Comment (wmalpica) [inline engine/src/parser/expression_utils.cpp:1504]: Do you know why `convert_concat_expression_into_multiple_binary_concat_ops` causes an error for e2e tests? Is there a particular query that is problematic? What happens to the expression here that makes it be a problem? Could it be that the issue is more that `convert_concat_expression_into_multiple_binary_concat_ops` does not properly handle some edge cases?

File: tests/BlazingSQLTest/EndToEndTests/TestSuites/nullaryOpsSuite.yaml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    APPLY_ORDER: True
+    USE_PERCENTAGE: False
+    ACCEPTABLE_DIFFERENCE: 0.01
+    COMPARE_WITH: drill
+    SKIP_WITH:
+      - JSON
+      - CSV
+      - CUDF
+
+  TEST_01: # current_date is not implemented as a nullaryOps, but need to be considered
```
Comment (wmalpica) [inline tests/BlazingSQLTest/EndToEndTests/TestSuites/nullaryOpsSuite.yaml:15]: Is current_date and current_timestamp something that is not supported by drill or pyspark? why do we have to validate it this way?

**Review by wmalpica — COMMENTED** [review]
I have some questions. I get the feeling that there is an issue in our expression modification code that we are sidestepping rather than fixing the root cause

File: engine/src/parser/expression_utils.cpp (hunk 1 above)
Comment (gcca) [reply engine/src/parser/expression_utils.cpp:1504]: Indeed, I updated this branch with the last fixes for concat-expressions. I commented `convert_concat_expression_into_multiple_binary_concat_ops` to indentify the issue source. Now I'm adapting it to cover the edge-cases.

File: tests/BlazingSQLTest/EndToEndTests/TestSuites/nullaryOpsSuite.yaml (hunk 2 above)
Comment (gcca) [reply tests/BlazingSQLTest/EndToEndTests/TestSuites/nullaryOpsSuite.yaml:15]: The e2e framework has the _mode_ `full` to perform the queries using drill or pyspark. But, that mode cannot be executed in gpuci. So we need a way to get result-queries for cases like `select current_date from ...` when the result need to be computed.
There is a task to implement that. This changes are a first approach.

Comment (wmalpica) [reply tests/BlazingSQLTest/EndToEndTests/TestSuites/nullaryOpsSuite.yaml:15]: For `select current_date` i can see how you cant do that in the CI framework using previously calculated results, but that is not what you have here. Here you can use previously calculated results

Comment (gcca) [reply tests/BlazingSQLTest/EndToEndTests/TestSuites/nullaryOpsSuite.yaml:15]: Indeed, it is. Than those queries because I found an error in the e2e: the result of the query changes if ORC or PARQUET is used for the creation of the table.

However, I have added the original queries again, since I was able to reproduce the error using the following script.

```python
from blazingsql import BlazingContext
import cudf

bc = BlazingContext()

bc.create_table('orders_parquet',
    ['.../blazingsql-testing-files/data/tpch/orders_0_0.parquet',
     '.../blazingsql-testing-files/data/tpch/orders_1_0.parquet']
)

bc.create_table('orders_orc', 
    ['.../blazingsql-testing-files/data/tpch/orders_0_0.orc',
     '.../blazingsql-testing-files/data/tpch/orders_1_0.orc']
)

print('>>> creado')

def run(tableName):
  query = f'''select o_orderdate
    from {tableName}
    where cast(o_orderdate as date) < CURRENT_DATE order by o_orderdate limit 5'''

  print('>>> explain')
  print(bc.explain(query))

  print('>>> sql query')
  df = bc.sql(query)

  print('>>> result')
  print(df)
  print(df.dtypes)

print('++++++++++++++++++++++++++++++++++++++++')
print('CON PARQUET')
print('++++++++++++++++++++++++++++++++++++++++')
run('orders_parquet')
print('\n++++++++++++++++++++++++++++++++++++++++')
print('CON ORC')
print('++++++++++++++++++++++++++++++++++++++++')
run('orders_orc')```

**Comment by felipeblazing** [comment]
rerun tests
